src/hotspot/share/services/diagnosticCommand.cpp

```diff
@@ -1113,7 +1130,13 @@ ThreadDumpToFileDCmd::ThreadDumpToFileDCmd(outputStream* output, bool heap) :
 
 void ThreadDumpToFileDCmd::execute(DCmdSource source, TRAPS) {
   bool json = (_format.value() != nullptr) && (strcmp(_format.value(), "json") == 0);
-  char* path = _filepath.value();
+  char path[JVM_MAXPATHLEN];
+  if (!Arguments::copy_expand_pid(_filepath.value(), strlen(_filepath.value()),
+                                  path, JVM_MAXPATHLEN)) {
+    stringStream msg;
+    msg.print("Invalid file path name specified: %s", _filepath.value());
+    THROW_MSG(vmSymbols::java_lang_IllegalArgumentException(), msg.base());
```

Comment:
write to output() and return instead of throwing

Reply:
Yes, the error needs to be written the stream so that it is printed by the tool (at the end other of the pipe).